macros/src/main/scala-2.11+/pureconfig/Derivation.scala

```diff
@@ -0,0 +1,196 @@
+package pureconfig
+
+import scala.collection.mutable
+import scala.language.experimental.macros
+import scala.reflect.macros.whitebox
+
+import pureconfig.derivation._
+
+sealed trait Derivation[A] {
+  def value: A
+}
```

Comment:
I didn't do it because the companion object and the trait have to be in the same file, and the companion objects of the two versions are different. Moreover, I wouldn't be able to make the trait `sealed` (unless I also defined `Failed` in 2.10).